query_builders/query_builder.py

```diff
@@ -591,21 +605,52 @@ def generate_query_body(
         :return: match string
         """
         assert match in ["MATCH", "OPTIONAL MATCH"]
+        assert tags is None or len(tags) == len(labels)   
+        for label in labels:
+            assert isinstance(label, str) or isinstance(label, dict)
+                
+        labels_str = [l.get('label') if isinstance(l, dict) else l for l in labels]
```

Comment:
At this point here, if `use_shortlabel=True` in `get_data_generic()` then all the labels (dict or str type) will be short labels. So we need to handle that here as well as our tests have `use_shortlabel=True` set.

Currently we get long labels from `generate_query_body()` even with  `use_shortlabel=True` set in data_provider!